### PR TITLE
Add schedulers and addable dictionary.

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

File: ranzen/misc.py [hunk 1]
```diff
@@ -147,3 +160,88 @@ def _generate_next_value_(name: str, start: int, count: int, last_values: list[A
...
+        ...
+
+    def __radd__(self: Self, other: int | dict[_KT, _VT]) -> Self | AddDict[_KT, _VT | list[_VT]]:
+        return self + other
+
+
+A = TypeVar("A", bound=Addable)
+
+
+def reduce_add(__iterable: Iterable[A]) -> A:
```
Comment (thomkeh) [inline ranzen/misc.py:238]: Why the double underscore? If you wanted to make the function work with positional arguments only, you can use `/`

Comment (MylesBartlett) [reply ranzen/misc.py:238]:
```
@overload
def sum(__iterable: Iterable[_SupportsSumNoDefaultT]) -> _SupportsSumNoDefaultT | Literal[0]: ...
```

Comment (thomkeh) [reply ranzen/misc.py:238]: Huh, why is that so weird?

**Comment by olliethomas** [comment]
what should happen for 
```
d1 = {"foo": [1, 11, 111]}
d2 = {"foo": 5}
d3 = d1 + d2
print(d3)
```

I think that the list concatenation should take priority and `d3 == {"foo": [1, 11, 111, 5]}`, but i guess it's reasonable that `d3 == {"foo": [6, 16, 116]}`.

File: ranzen/misc.py [hunk 2]
```diff
@@ -147,3 +160,89 @@ def _generate_next_value_(name: str, start: int, count: int, last_values: list[A
...
+        for key_o, value_o in other.items():
+            if key_o in self:
+                value_s = self[key_o]
+                if isinstance(value_s, Addable) and isinstance(value_o, Addable):
+                    try:
+                        copy[key_o] = value_s + value_o
+                    except TypeError:
+                        copy[key_o] = [value_s, value_o]
+                else:
+                    copy[key_o] = [value_s, value_o]
```
Comment (thomkeh) [inline ranzen/misc.py:219]: I feel like it's a bit surprising that it falls back to creating a list with two elements.

Take this code for example:
```python
class NotAddable:
    def __init__(self, x): self.x = x
d1 = {"foo": NotAddable(1)}
d2 = {"foo": NotAddable(2)}
d3 = {"foo": NotAddable(3)}
d12 = d1 + d2
d12 == {"foo": [NotAddable(1), NotAddable(2)]}
d12 + d3 == {"foo": [
                        [NotAddable(1), NotAddable(2)],
                        NotAddable(3),
                    ]}
```

Comment (thomkeh) [reply ranzen/misc.py:219]: I would just throw an error

Comment (MylesBartlett) [reply ranzen/misc.py:219]: Hmm maybe, or we check whether the fallback has already been invoked (checking for list+non-addable elements) and append in such instances

Comment (MylesBartlett) [reply ranzen/misc.py:219]: Though of course this fallback is also triggered by two addable types not being mutually addable...

File: pyproject.toml [hunk 3]
```diff
@@ -41,7 +41,6 @@ packages = [
 ]
 include=["ranzen/py.typed"]
 classifiers=[
-  "Programming Language :: Python :: 3.7",
```
Comment (thomkeh) [inline pyproject.toml:44]: Restore

File: ranzen/misc.py [hunk 4]
```diff
@@ -147,3 +160,110 @@ def _generate_next_value_(name: str, start: int, count: int, last_values: list[A
...
+                # if isinstance(value_s, Addable) and isinstance(value_o, Addable):
+                try:
+                    # Values are mutually addable (but not necessarily of the same type).
+                    copy[key_o] = value_s + value_o
+                except TypeError as e:
+                    msg = (
+                        f"Values of type '{type(value_s)}' and '{type(value_o)}' for key "
+                        "'{key_o}' are not mutuablly addable."
+                    )
+                    raise Exception(msg) from e
```
Comment (thomkeh) [inline ranzen/misc.py:234]: You might need to raise a typeerror

**Pull request merged**